### ISSUE TITLE
C-o not working on many non-vim commands/jumps

### ISSUE THREAD

**Issue by henrikrudstrom** [VSCodeVim/Vim#3990]
**Describe the bug**
`C-o` does not jump back after searching for refences or jumping to another location in another file through the explore etc.

**To Reproduce**
1. put the cursor over a function
2. right click and click `Find all references`
3. click a location on in the results view
4. press `C-o` 

also using pgup/pgdown/home does got get registered by the jumphandler

**Expected behavior**
expected to be returned back to the function definition, but instead it goes back to a previous jump i made.
also using pgup/pgdown/home does got get registered by the jumphandler

Is there a particular reason why we dont use vscode builtin `Go Back` / `Go Forward` commands? `workbench.action.navigateBack` / `workbench.action.navigateBack`

**Comment by summerhoum** [comment]
I have the same problem, for example:
I use "go to definition in extension c/c++" to  jump from functionA to functionB, and then jump to functionC. But When I use "ctrl + o" to go back, it will return to functionA, i think it should return to functionB.

**Comment by fabdolhosseini-xnor** [comment]
Have you tried the native VSCode "go back" command? It's mapped to `Ctrl + Alt + -` for me. Not sure if this is the right behaviour that you want, but a workaround is simply to change the keybindings and use this command instead.

**Comment by kidlj** [comment]
Native VS Code "go back" will record every CURSOR move, even "hjkl", which is not great.

The vim Ctrl+o is in a state of chaos.

**Comment by kidlj** [comment]
Native VSCode "go back" will record every CURSOR move, even "hjkl" keystrokes, which is not great to use.

Vim Ctrl+o is in a state of chaos.

**Comment by kidlj** [comment]
I can confirm that version 1.11.3 works well on jumping forward & backward across different files.

**Comment by yeonsh** [comment]
"Go back" does not work on version 1.12.4 after using "go to definition" on C/C++ files.

**Comment by Hubro** [comment]
One year later, ctrl+o still does not work after "Jump to definition". Ctrl+Alt+- does work though, but it's very distracting to try to keep in mind when to use each type of "go back" command.

**Comment by Hubro** [comment]
I have now mapped ctrl+o and ctrl+i to the native vscode backwards/forwards commands:

```
    {
        "key": "ctrl+o",
        "command": "workbench.action.navigateBack"
    },
    {
        "key": "ctrl+i",
        "command": "workbench.action.navigateForward"
    },
```

So far this appears to work far better than the default bindings.

**Comment by J-Fields** [comment]
@Hubro How are you invoking `Jump to definition`?

**Comment by Hubro** [comment]
> @Hubro How are you invoking `Jump to definition`?

Usually just Ctrl+Click. By your question I'm guessing there's a Vim mode binding for `Jump to definition` that does work with ctrl+i/o by default?

**Comment by J-Fields** [comment]
> Usually just Ctrl+Click. By your question I'm guessing there's a Vim mode binding for `Jump to definition` that does work with ctrl+i/o by default?
> 
> EDIT: Yup, the `gd` command (that I've unfortunately never heard of before now) does go-to-definition, and it works with the default ctrl+i/o.

Yep, `<C-]>` also does the same. The extension isn't told that you've done a `Jump to definition` with a mouse click, and we *might* be able to make an educated guess, but it's complicated and that sort of thing isn't currently implemented.

**Comment by Hubro** [comment]
> Yep, `<C-]>` also does the same. The extension isn't told that you've done a `Jump to definition` with a mouse click, and we _might_ be able to make an educated guess, but it's complicated and that sort of thing isn't currently implemented.

Any idea why the Vim extension doesn't just use VSCode's native/built-in stack of movements/jumps rather than implementing its own? Sounds like it would be simpler.

**Comment by J-Fields** [comment]
> > Yep, `<C-]>` also does the same. The extension isn't told that you've done a `Jump to definition` with a mouse click, and we _might_ be able to make an educated guess, but it's complicated and that sort of thing isn't currently implemented.
> 
> Any idea why the Vim extension doesn't just use VSCode's native/built-in stack of movements/jumps rather than implementing its own? Sounds like it would be simpler.

It's been considered. The main problem is that for commands like `hjkl` for instance, we just tell VSCode "the new cursor position is <whatever>" and this actually triggers a jump in its list. We could get around this by using the `moveCursor` command, but it has performance implications IIRC. Definitely something that requires more thought on my part - maybe if I asked nicely enough the VSCode team would add a new API to allow setting the cursor position directly without triggering a jump.

**Comment by jgarciaokode** [comment]
I noticed <C-o> works well when using "gd" go to definition. However, when you use vscode's command revealDefinitionAside, <C-o> is not working properly, i guess the problem is that it's not registering the jump action. It would be awesome if it worked like go to definition.

    {
      "before": ["g", "v"],
      "commands": ["editor.action.revealDefinitionAside"],
      "when": "editorHasDefinitionProvider && editorTextFocus && !isInEmbeddedEditor"
    },
